server/src/main/java/org/elasticsearch/index/shard/IndexShard.java

```diff
@@ -2301,13 +2309,21 @@ public void onFailure(final Exception e) {
                     }
                 },
                 executorOnDelay,
-                true);
+                true, debugInfo);
     }
 
     public int getActiveOperationsCount() {
         return indexShardOperationPermits.getActiveOperationsCount(); // refCount is incremented on successful acquire and decremented on close
     }
 
+    /**
+     * @return a list of containing an exceptions for each operation permit that wasn't released yet. The stack traces of the exceptions
```

Comment:
there seems to be a missing word

Reply:
There's an unneeded s in exceptions and a missing s in contain. Can you clarify?

Reply:
ok, I thought it meant to say "a list of **items/things/XYZ** containing an exception ..."